проект изучение иностранных слов

Leksa2304 opened the pull request:
реализовано:
выборка из 5 неповторяющихся слов,
слайдер, переход в режим тестирования, отображение перемешанных карточек

Leksa2304 committed:
проект изучение иностранных слов

реализовано:
выборка из 5 неповторяющихся слов,
слайдер, переход в режим тестирования, отображение перемешанных карточек

Leksa2304 committed:
замена на создание одной карточки

удалено создание карточек createExamCardEng и createExamCardRu. Замена на создание createExamCard.

Leksa2304 committed:
add обработчик на первый клик по карточке

для окрашивания ее в зеленый (класс .correct)

Leksa2304 committed:
add переменная на 5 слов и change названия

Leksa2304 committed:
add f блокировки кнопок

Leksa2304 committed:
работает кнопка Перемешать слова

добавлено отображение слова (ShowCard)

Leksa2304 committed:
add при создании exam карточек атрибуты

и корректировка в f вставки карточек

Leksa2304 committed:
add проверка слов в режиме тестирования

откорректирована проверка на первый клик

Leksa2304 committed:
add тестирование успешно завершено

Leksa2304 committed:
add таймер

Leksa2304 committed:
add окно Твоя статистика

после завершения тестирования с указанием время тестирования

Leksa2304 committed:
add шаблона статистики

Leksa2304 committed:
add кнопка Назад к тренировке

реализован выход из режима экзамена для повторения слов

Leksa2304 committed:
add отображение % отвеченных слов

Leksa2304 committed:
add прогресс бар в режиме экзамена

Leksa2304 committed:
change подсчет % правильных ответов exam

код вынесен в функцию

Leksa2304 committed:
add img по завершению тестирования

Leksa2304 committed:
add прогресс бар для режима тренировки

add функция прогресс бара назад

Leksa2304 committed:
change в создании дополнительных кнопок

отдельная функция для создания кнопки

Leksa2304 committed:
change запуск экзамена вынесен в отдельную функцию

и добавлен обработчик на дополнительную кнопку Перезапустить экзамен

Leksa2304 committed:
chang let на  const

Leksa2304 committed:
corrected на срабатывание дополнительных кнопок

нельзя нажать, пока видна картинка, т.к окно статистика не доделано и  имеет небольшой размер. Еще переименование кнопок на buttonStadyAgain и buttonExamAgain

Leksa2304 committed:
change названия функций

и добавлен отступ кнопке

Leksa2304 committed:
change отступ кнопке

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,383 @@
+const percent = 100;
+let correctWords = 0; // счетчик верных ответов
+const percentOneCard = percent / maxWords; // процент верного ответа одной карточки
+let correctPercent = document.querySelector("#correct-percent");
```
Лена,  сама переменная не перезаписывается. Будет const. То,  что меняется textContent  -  это происходит как бы "внутри"

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,383 @@
+
+function getRandomWord(arr) { // функция для генерации карточки (случайного объекта из массива)
+    const index = Math.floor(Math.random() * arr.length);
+    let obj = arr[index]; // получаем случайный объект
```
Почему let?) я не нашла,  чтобы было перезаписано как 
obj  = что-нибудь

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,383 @@
+    spanExample.innerHTML = `${card.example}`;
+}
+
+let wordsLearning = []; // массив слов для изучения
```
и снова const . Всё, что потом происходит с wordsLearning   -  это "внутри", но сама переменная явно не перезаписывается

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,383 @@
+        }, 1000);
+    }
+    timer.textContent = `${time.textContent}`;
+};
```
А вот тут, Лена, есть баг. Выбери первую пару,  она исчезает. А потом кликни на другую карточку и потом на пустое место(где карточка исчезла ). Или сначала на пустое место,  а потом на любую другую карточку. Или 2 раза на пустое место. 

Это нужно пофиксить)

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,383 @@
+
+        setTimeout(() => {
+            alert("Тестирование успешно пройдено!");
+            showImg();
```
Лена,  у тебя картинка вставилась после подключения скрипта
![Screenshot_138](https://github.com/Leksa2304/foreign-words-training/assets/24245326/acdc7b84-b298-4220-843e-3f69eae79c63)
 и очень плохая идея делать делать ей absolut, с ним вообще поаккуратнее. Лучше найди для неё место и вставь

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,383 @@
+    button.style.marginBottom = "10px";
+    container.append(button);
+
+}
```
Тут для тренировки ок,  но если брать реальный проект  -  то просто добавляем в разметку что нужно,  чтобы не трать на это ресурс)

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,383 @@
+const buttonExamAgain = document.querySelector("#button-exam-again");
+
+
+function resetExamMode() { // сброс режима экзамена
```
Насчет комментов  -  опять же для тренировки норм,  но в реальных проектах  оставляем только нужные комменты(например что-то описываем,  чтобы второй разраб понял) , или то, что требуется в проекте. Всё остальное удаляем

Leksa2304 committed:
corrected let на const, кнопки перенесены в разметку html

Leksa2304 committed:
change вернула создание доп. кнопок в jsБ удалила  из html  и style

Leksa2304 committed:
corrected после исчезновения карточек они неактивны

Leksa2304 committed:
change img после окончания экзамена

img вынесено в разметку html

Leksa2304 committed:
change stat

Leksa2304 committed:
add в статистику слова без учета количества попыток

Leksa2304 committed:
add в статистику количество попыток

код работает

Leksa2304 committed:
corrected кнопки Назад к тренировке

чтобы корректно работал прогресс бар и счетчик (сброс на первое слово). Функции вынесены вверх кода.

Leksa2304 committed:
optimization кода

отдельные функции для удаления/добавления видимости
